From c3ccc36e94f0d7bfd489f8ff66dd2be0ce9fca4f Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Wed, 26 Mar 2025 14:14:20 +0100
Subject: [PATCH 1661/1695] drivers: stm32_tamp: call
 stm32_tamp_set_secret_list() after applying rif configuration

The function stm32_tamp_set_secret_list() access to register
TAMP_ERCFGR.
TAMP_ERCFGR access rights are linked to the configuration written in
TAMP_R0CIDCFGR. Before accessing TAMP_ERCFGR we need to set config from
the DT in TAMP_R0CIDCFGR in case the current configuration is invalid.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git edc6ce094fe581ea22755b265f07697ad53d43c8 drivers: stm32_tamp: call stm32_tamp_set_secret_list() after applying rif configuration]
Change-Id: I97d7abcd5b12d02b8898fccf84db337920a918a2
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/446775
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_tamp.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/core/drivers/stm32_tamp.c b/core/drivers/stm32_tamp.c
index de014525d..d338bf7b2 100644
--- a/core/drivers/stm32_tamp.c
+++ b/core/drivers/stm32_tamp.c
@@ -2068,14 +2068,6 @@ static TEE_Result stm32_tamp_probe(const void *fdt, int node,
 		stm32mp_syscfg_write(SYSCFG_POTTAMPRSTCR, BIT(0), BIT(0));
 #endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21) */
 
-	/*
-	 * Select extra IP to add in the deleted/blocked IP in case of
-	 * tamper event
-	 *
-	 * No IP added.
-	 */
-	stm32_tamp_set_secret_list(&stm32_tamp, 0);
-
 	if (stm32_tamp.pdata.compat->tags & TAMP_HAS_RIF_SUPPORT) {
 		apply_rif_config();
 
@@ -2106,6 +2098,14 @@ static TEE_Result stm32_tamp_probe(const void *fdt, int node,
 					 _TAMP_PRIVCFG_BKPWPRIV);
 	}
 
+	/*
+	 * Select extra IP to add in the deleted/blocked IP in case of
+	 * tamper event
+	 *
+	 * No IP added.
+	 */
+	stm32_tamp_set_secret_list(&stm32_tamp, 0);
+
 	if (!(stm32_tamp.pdata.compat->tags & TAMP_HAS_RIF_SUPPORT) ||
 	    stm32_tamp.pdata.is_tdcid) {
 		res = stm32_tamp_set_secure_bkpregs();
-- 
2.47.2

